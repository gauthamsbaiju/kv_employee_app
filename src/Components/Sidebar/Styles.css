.sidebar{
    background-color: white;
    position: fixed ;
    top: 77px;
    left: 0px;
    width: 300px;
    height: 1019px;
}

.btn-wrapper{
    display: flex;
    align-items:center;
    background-color: #03aeee;
    height: 60px;
    margin-left: 7%;
    margin-top: 9%;
    border-radius: 50px 0px 0px 50px;
}

.btn-icon{
    background-color: white;
    border-radius: 50%;
    margin-left: 15px;
    height: 29px;
    width: 29px;
    display: flex;
    align-items:center;
    justify-content: center;

}

.btn-text{
    height: 22px;
    width: 127px;
    color: white;
    margin-left: 12px;
    font-size: 16px;
    font-weight:600;
}