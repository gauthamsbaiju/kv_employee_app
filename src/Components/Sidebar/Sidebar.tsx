import React from 'react';
import './Styles.css';
import { useNavigate } from 'react-router-dom';

// type InputPropsType = {
//   label: string;
//   className: string;
//   onClick: (e) => void;
//   type: 'submit';
// };

const Sidebar: React.FC = () => {
  const navigate = useNavigate();
  const handleClick = () => {
    navigate(`/employee`);
  };

  return (
    <div className='sidebar'>
      <div className='btn-wrapper' onClick={handleClick}>
        <div className='btn-icon'>
          <img src='/assets/icons/employees.svg' alt='icon' className='icon'></img>
        </div>
        <div className='btn-text'>Employee List</div>
      </div>
    </div>
  );
};

export default Sidebar;
