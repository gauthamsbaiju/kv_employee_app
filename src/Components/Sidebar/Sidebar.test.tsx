import { render, screen } from '@testing-library/react';
import Sidebar from './Sidebar';
import '@testing-library/jest-dom';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate
}));

describe('Test for sidebar component', () => {
  test('Test for snapshot rendered', () => {
    render(<Sidebar />);
    const element = screen.getByTestId('sidebar-test');

    expect(element).toMatchSnapshot();
  });

  test('Test for rendered elements', () => {
    render(<Sidebar />);
    const element = screen.getByTestId('sidebar-test');

    expect(element).toMatchSnapshot();
  });
});
