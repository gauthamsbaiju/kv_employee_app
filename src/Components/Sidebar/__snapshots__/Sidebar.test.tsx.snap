// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for sidebar component Test for snapshot rendered 1`] = `
<div
  class="sidebar"
  data-testid="sidebar-test"
>
  <div
    class="btn-wrapper"
  >
    <div
      class="btn-icon"
    >
      <img
        alt="icon"
        class="icon"
        src="/assets/icons/employees.svg"
      />
    </div>
    <div
      class="btn-text"
    >
      Employee List
    </div>
  </div>
</div>
`;
